Cannot start docker engine after upgrade of both fedora & docker-engine due to selinux errors

Previously running Fedora 23 with docker-engine{,-selinux} 1.10.3 from Docker repos (not Fedora repos) - working fine for months. Upgraded to Fedora 24, updated to Docker 1.11.2 throws an error:

```
Re-declaration of boolean virt_sandbox_use_fusefs
Failed to create node
Bad boolean declaration at line 148 of /var/lib/selinux/targeted/tmp/modules/100/virt/cil
/usr/sbin/semodule:  Failed!
```

Have a colleague with exactly the same problem. Tried uninstalling RPMs & reinstalling but no luck.

**Output of `docker version`:**

```
Client:
 Version:      1.11.2
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   b9f10c9
 Built:        Wed Jun  1 21:31:13 2016
 OS/Arch:      linux/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

**Output of `docker info`:**

```
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Laptop running Fedora 24

**Steps to reproduce the issue:**
Not sure it's 100% reproducible, but have a couple of colleagues with the same issue.

**Describe the results you received:**
Cannot start docker.

**Describe the results you expected:**
Able to start docker.

@crosbymichael the build chugs along for a while, and then I get this:

    ---> Making bundle: build-rpm (in bundles/1.12.0-dev/build-rpm)
    INFO: Waiting for daemon to start...
    +++ exec dockerd --debug --host unix:///go/src/github.com/docker/docker/bundles/1.12.0-dev/build-rpm/docker.sock --storage-driver devicemapper --pidfile bundles/1.12.0-dev/build-rpm/docker.pid --userland-proxy=true
    ...........................................................
    error: daemon at unix:///go/src/github.com/docker/docker/bundles/1.12.0-dev/build-rpm/docker.sock fails to 'docker version':
    Client:
     Version:      1.12.0-dev
     API version:  1.25
     Go version:   go1.6.2
     Git commit:   80ec9fa
     Built:        Fri Jul 15 15:35:51 2016
     OS/Arch:      linux/amd64
    Cannot connect to the Docker daemon. Is the docker daemon running on this host?
    ---> Making bundle: .integration-daemon-stop (in bundles/1.12.0-dev/build-rpm)
    Makefile:100: recipe for target 'rpm' failed

This is my fork of docker based on `upstream/master`

--- a/contrib/init/systemd/docker.service.rpm
+++ b/contrib/init/systemd/docker.service.rpm
@@ -1,0 +1,29 @@
+[Unit]
+Description=Docker Application Container Engine
+Documentation=https://docs.docker.com
+After=network.target
+Requires=docker.socket
+
+[Service]
+Type=notify
+# the default is not to use systemd for cgroups because the delegate issues still
+# exists and systemd currently does not support the cgroup feature set required
+# for containers run by docker
+ExecStart=/usr/bin/dockerd
+ExecReload=/bin/kill -s HUP $MAINPID
+# Having non-zero Limit*s causes performance problems due to accounting overhead
+# in the kernel. We recommend using cgroups to do container-local accounting.
+LimitNOFILE=infinity
+LimitNPROC=infinity
+LimitCORE=infinity
+# Uncomment TasksMax if your systemd version supports it.
+# Only systemd 226 and above support this version.
+#TasksMax=infinity
+TimeoutStartSec=0
+# set delegate yes so that systemd does not reset the cgroups of docker containers
+Delegate=yes
+# kill only the docker process, not all processes in the cgroup
+KillMode=process
+
+[Install]
+WantedBy=multi-user.target

--- a/docs/installation/linux/fedora.md
+++ b/docs/installation/linux/fedora.md
@@ -60,9 +60,9 @@ There are two ways to install Docker Engine.  You can install with the `dnf` pac
 
         $ sudo dnf install docker-engine
 
-5. Enable the socket and service.
+5. Enable the service.
 
-		$ sudo systemctl enable docker.socket docker.service
+		$ sudo systemctl enable docker.service
 
 6. Start the Docker daemon.
 
@@ -113,9 +113,9 @@ There are two ways to install Docker Engine.  You can install with the `dnf` pac
 
 	This script adds the `docker.repo` repository and installs Docker.
 
-4. Enable the socket and service.
+4. Enable the service.
 
-		$ sudo systemctl enable docker.socket docker.service
+		$ sudo systemctl enable docker.service
 
 5. Start the Docker daemon.
 

--- a/hack/make/.build-rpm/docker-engine.spec
+++ b/hack/make/.build-rpm/docker-engine.spec
@@ -147,8 +147,7 @@ install -d $RPM_BUILD_ROOT/%{_initddir}
 
 %if 0%{?is_systemd}
 install -d $RPM_BUILD_ROOT/%{_unitdir}
-install -p -m 644 contrib/init/systemd/docker.service $RPM_BUILD_ROOT/%{_unitdir}/docker.service
-install -p -m 644 contrib/init/systemd/docker.socket $RPM_BUILD_ROOT/%{_unitdir}/docker.socket
+install -p -m 644 contrib/init/systemd/docker.service.rpm $RPM_BUILD_ROOT/%{_unitdir}/docker.service
 %else
 install -p -m 644 contrib/init/sysvinit-redhat/docker.sysconfig $RPM_BUILD_ROOT/etc/sysconfig/docker
 install -p -m 755 contrib/init/sysvinit-redhat/docker $RPM_BUILD_ROOT/%{_initddir}/docker
@@ -194,7 +193,6 @@ install -p -m 644 contrib/syntax/nano/Dockerfile.nanorc $RPM_BUILD_ROOT/usr/shar
 /%{_sysconfdir}/udev/rules.d/80-docker.rules
 %if 0%{?is_systemd}
 /%{_unitdir}/docker.service
-/%{_unitdir}/docker.socket
 %else
 %config(noreplace,missingok) /etc/sysconfig/docker
 /%{_initddir}/docker

